e2e/cypress/integration/accessibility/accessibility_modals_dialogs_spec.js

```diff
@@ -0,0 +1,220 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+let selectedRowText;
+
+function verifyMainMenuModal(modalName, modalId, modalLabel, expectedModalName) {
+    cy.get('#headerInfo button').click();
+    cy.findByText(modalName).click();
+    cy.get(`#${modalId}`).should('have.attr', 'role', 'dialog').and('have.attr', 'aria-labelledby', modalLabel).within(() => {
```

Comment:
I wonder if it's possible to make use of [ByRole](https://testing-library.com/docs/dom-testing-library/api-queries#byrole) and [ByLabelText](https://testing-library.com/docs/dom-testing-library/api-queries#bylabeltext).

Reply:
There are actually multiple elements with the role dialog in the page and hence I had to use the unique ID of the element.